src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Store.cs

```diff
@@ -114,12 +173,39 @@ public void Remove(X509Certificate2 certificate)
             _storePal.Remove(certificate.Pal);
         }
 
+        public void RemoveRange(X509Certificate2Collection certificates)
+        {
+            if (certificates == null)
+                throw new ArgumentNullException(nameof(certificates));
+
+            int i = 0;
+            try
+            {
+                foreach (X509Certificate2 certificate in certificates)
+                {
+                    Remove(certificate);
+                    i++;
+                }
+            }
+            catch
+            {
+                // For desktop compat, we keep the exception semantics even though they are not ideal
+                // because a missing cert will raise exception causing certs to be added even if 
+                // they weren't there before.
+                for (int j = 0; j < i; j++)
+                {
+                    Add(certificates[j]);
```

Comment:
Nothing. The code is broken in exception cases but left as-is for netfx compat. If we diverge, we should probably change netfx also. Another option is to throw PNSE for these two methods.